index.d.ts

```diff
@@ -63,67 +68,59 @@ export type Logger = (level: string, ...args: any[]) => void;
 
 export type Emit<T> = (input: T) => void;
 
-export interface SagaMiddlewareOptions<C extends object> {
+export interface SagaMiddlewareOptions<C extends object = {}> {
   context?: C;
   sagaMonitor?: Monitor;
   logger?: Logger;
   onError?(error: Error): void;
+  effectMiddlewares?: EffectMiddleware[];
   emitter?(emit: Emit<Action>): Emit<any>;
 }
 
 export default function sagaMiddlewareFactory<C extends object>(
   options?: SagaMiddlewareOptions<C>,
 ): SagaMiddleware<C>;
 
-
-type Unsubscribe = () => void;
-type Subscribe<T> = (cb: (input: T | END) => void) => Unsubscribe;
-
 export interface RunSagaOptions<A, S> {
-  context?: object;
-  subscribe?: Subscribe<A>;
+  channel?: PredicateTakeableChannel<A>;
```

Comment:
While technically this is correct, because saga will allow for it - the best option is to pass `stdChannel` here, maybe typings could enforce it? Dunno what best practices are in such cases.

Reply:
We can use `MulticastChannel` type here, though that wouldn't enforce exactly `stdChannel`.

Reply:
Hm, probably not worth it then. I guess that somebody might find it useful to pass in a unicast channel anyway, so it's better to not strangle one's creativity.